### PR TITLE
[Improvement][Alert] Add timeout params for HTTP plugin

### LINKED ISSUE #15172: [Improvement][Alert] Add timeout params for HTTP plugin
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Add timeout params for HTTP plugin

### Are you willing to submit a PR?

- [X] #15173

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by qingwli**
- close #15172

### CODE REVIEW

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-dingtalk/src/main/java/org/apache/dolphinscheduler/plugin/alert/dingtalk/DingTalkSender.java [hunk 1]
```diff
@@ -109,12 +108,11 @@ private static CloseableHttpClient getProxyClient(String proxy, int port, String
         HttpHost httpProxy = new HttpHost(proxy, port);
         CredentialsProvider provider = new BasicCredentialsProvider();
         provider.setCredentials(new AuthScope(httpProxy), new UsernamePasswordCredentials(user, password));
-        return HttpClients.custom().setRetryHandler(HttpServiceRetryStrategy.retryStrategy)
```
Comment (ruanwenjun) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-dingtalk/src/main/java/org/apache/dolphinscheduler/plugin/alert/dingtalk/DingTalkSender.java]: You will loss CredentialsProvider here?

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-api/src/main/java/org/apache/dolphinscheduler/alert/api/HttpUtils.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                .setConnectTimeout(TIMEOUT_MILLIS)
+                .setConnectionRequestTimeout(TIMEOUT_MILLIS)
+                .setSocketTimeout(TIMEOUT_MILLIS)
+                .build();
+
+        return HttpClients.custom()
+                .setDefaultRequestConfig(requestConfig)
+                .setRetryHandler(HttpServiceRetryStrategy.retryStrategy).build();
+    }
+}
```
Comment (ruanwenjun) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-api/src/main/java/org/apache/dolphinscheduler/alert/api/HttpUtils.java]: I am not suggest to provide a basic client utils here, different plugin may have different client config.

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-http/src/main/java/org/apache/dolphinscheduler/plugin/alert/http/HttpSender.java [hunk 3]
```diff
@@ -108,8 +113,15 @@ public AlertResult send(String msg) {
     }
 
     public String getResponseString(HttpRequestBase httpRequest) throws IOException {
-        CloseableHttpClient httpClient =
-                HttpClients.custom().setRetryHandler(HttpServiceRetryStrategy.retryStrategy).build();
+        RequestConfig requestConfig = RequestConfig.custom()
```
Comment (ruanwenjun) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-http/src/main/java/org/apache/dolphinscheduler/plugin/alert/http/HttpSender.java:116]: Use TimeUnit to convert s to ms or directly use timeout * 1000 is ok, no need to define 
NUMBER_1000

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-http/src/main/java/org/apache/dolphinscheduler/plugin/alert/http/HttpAlertChannelFactory.java [hunk 4]
```diff
@@ -79,7 +79,15 @@ public List<PluginParams> params() {
                                 .build())
                         .build();
 
-        return Arrays.asList(url, requestType, headerParams, bodyParams, contentField);
+        InputParam timeout =
```
Comment (ruanwenjun) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-http/src/main/java/org/apache/dolphinscheduler/plugin/alert/http/HttpAlertChannelFactory.java]: Please use InputNumberParam` is better, and use `setValue1 to set default value to 120.

Comment (qingwli) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-http/src/main/java/org/apache/dolphinscheduler/plugin/alert/http/HttpAlertChannelFactory.java]: Done

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-http/src/main/java/org/apache/dolphinscheduler/plugin/alert/http/HttpSender.java (hunk 3 above)
Comment (qingwli) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-http/src/main/java/org/apache/dolphinscheduler/plugin/alert/http/HttpSender.java:116]: Done

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Pull request merged**